cpu/native/mtd/mtd_native.c

```diff
@@ -79,14 +79,16 @@ static int _write(mtd_dev_t *dev, const void *buff, uint32_t addr, uint32_t size
 {
     mtd_native_dev_t *_dev = (mtd_native_dev_t*) dev;
     size_t mtd_size = dev->sector_count * dev->pages_per_sector * dev->page_size;
-    size_t sector_size = dev->pages_per_sector * dev->page_size;
 
-    DEBUG("mtd_native: write from page %" PRIu32 " count %" PRIu32 "\n", addr, size);
+    DEBUG("mtd_native: write from 0x%" PRIx32 " count %" PRIu32 "\n", addr, size);
 
+    if (size > dev->page_size) {
```

Comment:
this is somewhat duplicated, the condition below `((addr % dev->page_size) + size) > dev->page_size` already covers those cases and both return codes are equal.

Reply:
Fixed